samples/ChatSample.Cli/Program.cs

```diff
@@ -0,0 +1,558 @@
+    };
+
+    private static readonly Dictionary<string, TypeInfo[]> MethodDefineMap = new(StringComparer.OrdinalIgnoreCase);
+
+    private static readonly Dictionary<string, (TypeInfo[] Params, TypeInfo Item)> StreamMethodDefineMap = new(StringComparer.OrdinalIgnoreCase);
+
+    private static readonly Dictionary<(string ConnectionId, string StreamId), (TypeInfo TypeInfo, Channel<object> Channel)> StreamMap = [];
+
+    private static async Task Main()
+    {
+        while (true)
+        {
+            var line = Console.ReadLine()?.Trim();
+            if (line == null)
+            {
+                return;
+            }
+            if (line.Length == 0)
+            {
+                continue;
+            }
+            if (line.Equals("exit", StringComparison.OrdinalIgnoreCase))
+            {
+                break;
+            }
+            if ("help".Equals(line, StringComparison.OrdinalIgnoreCase))
+            {
+                Console.WriteLine("Available commands:");
+                Console.WriteLine("  connect <connection-string> <hub> [--transient/-t]");
+                Console.WriteLine("  client <connection-string> <hub>");
+                Console.WriteLine("  help");
+                Console.WriteLine("  exit");
+                continue;
+            }
+            if ("list".Equals(line, StringComparison.OrdinalIgnoreCase))
+            {
+                foreach (var (m, p) in MethodDefineMap)
+                {
+                    Console.WriteLine($"Target: {m}, ArgTypes: {string.Join(", ", p.Select(x => x.Type.Name))}");
+                }
+                continue;
+            }
+            if (await Try(ConnectRegex(), line, Connect) ||
+                await Try(ClientRegex(), line, RunAsClient))
+            {
+                continue;
+            }
+
+            Console.WriteLine($"Unrecognized command: {line}");
+            Console.WriteLine("Type 'help' for a list of available commands.");
+        }
+    }
+
+    private static async ValueTask<bool> Try(Regex regex, string line, Func<Match, Task> action)
+    {
+        var match = regex.Match(line);
+        if (match.Success)
+        {
+            await action(match);
+            return true;
+        }
+        return false;
+    }
+
+    private static async Task Connect(Match match)
+    {
+        var connectionString = match.Groups[1].Value;
+        var hub = match.Groups[2].Value;
+        var transportType = match.Groups[3].Value.Length > 0 ? ServiceTransportType.Transient : ServiceTransportType.Persistent;
+        Console.WriteLine($"Connecting to {connectionString} for {hub} with transport type {transportType}");
+        await ConnectAsync(
+            connectionString,
+            hub,
+            transportType);
+    }
+
+    private static async Task ConnectAsync(string connectionString, string hub, ServiceTransportType transportType)
+    {
+        var serviceManager = new ServiceManagerBuilder().WithOptions(option =>
```

Comment:
Don't forget to dispose service manager and service hub context, to set a good example for our users。